joomla/joomla-cms
# [4.0] Fix debug logging. Move logging to application

This is try to fix  Release Blocker #25675 .

### Summary of Changes

This will move logging set up to application level. The plugin used only for display them.
One of new feature here, that the logging may be enabled even when the debugging is Off.


### Testing Instructions

Be aware: there a thousands of Deprecated entries, if you have a slow PC / server, be patient.

Apply patch. Go to Global Configuration and enable Debug, also enable Log Deprecated API.
Enable System Debug plugin, and set "Show logs" in the plugin.

In the debugger toolbar you should be able to see much of entries of Deprecated call.


### Actual result BEFORE applying this Pull Request

No Deprecated entries, or only few of them.


### Expected result AFTER applying this Pull Request

A loot of entries


### Documentation Changes Required

Yes I guess. Logging moved to Global configuration, and can be enabled there.
